controller/websocket_controller.go

```diff
@@ -0,0 +1,133 @@
+package controller
+
+import (
+	"sync"
+
+	"github.com/Sirupsen/logrus"
+
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/kubernetes/pkg/controller"
+
+	lhinformers "github.com/rancher/longhorn-manager/k8s/pkg/client/informers/externalversions/longhorn/v1alpha1"
+)
+
+type SimpleResourceEventHandler struct{ ChangeFunc func() }
+
+func (s SimpleResourceEventHandler) OnAdd(obj interface{})               { s.ChangeFunc() }
+func (s SimpleResourceEventHandler) OnUpdate(oldObj, newObj interface{}) { s.ChangeFunc() }
+func (s SimpleResourceEventHandler) OnDelete(obj interface{})            { s.ChangeFunc() }
+
+type Watcher struct {
+	eventChan  chan struct{}
+	resources  []string
+	controller *WebsocketController
+}
+
+func (w *Watcher) Events() <-chan struct{} {
+	return w.eventChan
+}
+
+func (w *Watcher) Close() {
+	wc := w.controller
+	wc.watcherLock.Lock()
+	defer wc.watcherLock.Unlock()
+
+	close(w.eventChan)
+	for i, watcher := range wc.watchers {
+		if watcher == w {
+			wc.watchers = append(wc.watchers[:i], wc.watchers[i+1:]...)
+			return
+		}
+	}
+	logrus.Warningf("failed to remove from watchers: %+v", w)
```

Comment:
Both eventChan and controller is meaningless for log here. Just show the resources in the log.